source/history.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var likely = require('./index.js');
+var utils = require('./utils');
+
+/**
+ * History
+ * @type {Object}
+ */
+var history = {
+    /**
+     * Called everytime the page url's been changed.
+     * Reinits all widgets with the new page url.
+     * @type {Function}
+     */
+    onUrlChanges: function () {
+        likely.initiate({
+            forceUpdate: true,
+            url: utils.getDefaultUrl(),
+        });
+    },
+    /**
+     * Inits pust/pop state events listeners
+     * @type {Function}
+     */
+    init: function () {
+        var pushState = window.history.pushState;
+        window.history.pushState = function () {
+            // browser should change the url first
+            setTimeout(this.onUrlChanges, 0);
+            return pushState.apply(window.history, arguments);
+        }.bind(this);
+
+        var replaceState = window.history.replaceState;
+        window.history.replaceState = function () {
+            setTimeout(this.onUrlChanges, 0);
+            return replaceState.apply(window.history, arguments);
+        }.bind(this);
+
+        window.addEventListener('popstate', this.onUrlChanges);
```

Comment:
Lost one more point, sorry. This should also be bound so that if we reference `this` from `onUrlChanges` it will work as intended.